### ISSUE TITLE
Traffic shaping (`bandwidth` CNI plugin) does not work with `cni: flannel`

### ISSUE THREAD

**Issue by brandond** [rancher/rke2#7244]
**Environmental Info:**
RKE2 Version: 
n/a

Node(s) CPU architecture, OS, and Version: 
linux

Cluster Configuration:
n/a

**Describe the bug:**
Cannot follow upstream instructions to enable traffic shaping:
https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#support-traffic-shaping

**Steps To Reproduce:**
1. Install RKE2 with flannel CNI
2. Deploy pod with bandwidth annotations

**Expected behavior:**
bandwith plugin is executed to set traffic limits on network namespace

**Actual behavior:**
Fails because rke2-flannel CNI config lacks bandwidth plugin config

**Additional context / logs:**
May also be an issue for other CNIs, I haven't actually tested.

**Comment by manuelbuil** [comment]
I think the issue is mostly about making both RKE2 and K3s config the same. There are certain "meta" cni plugins: https://github.com/containernetworking/plugins/tree/main/plugins/meta which you can stack together with the main cni plugin. But I don't think we should add all of them by default to network plugins

**Comment by brandond** [comment]
yeah but you need to enable bandwidth in the flannel config for the Kubernetes feature to work properly. Ref: https://github.com/k3s-io/k3s/pull/7425

**Comment by caroline-suse-rancher** [comment]
Hey @manuelbuil and @brandond any updates on this one?

**Comment by brandond** [comment]
probably still needs to be addressed in the flannel config for rke2-flannel and rke2-canal.

**Comment by rbrtbnfgl** [comment]
I'm adding a way on the flannel chart to edit the CNI conf file https://github.com/flannel-io/flannel/pull/2195 if we want the bandwidth CNI by default we can patch the value file on rke2-chart.

**Comment by rbrtbnfgl** [comment]
On Canal and Calico it's enabled by default. Cilium has its own bandwidth manager.

**Comment by rbrtbnfgl** [comment]
This is fixed with the latest Flannel update #7844

**Comment by fmoral2** [comment]
validated here:
https://github.com/rancher/rke2/issues/7871
https://github.com/rancher/rke2/issues/7872
https://github.com/rancher/rke2/issues/7873#issuecomment-2730033514
